newIDE/app/src/ObjectEditor/Editors/Model3DEditor.js

```diff
@@ -0,0 +1,269 @@
+  const properties = objectConfiguration.getProperties();
+
+  const onChangeProperty = React.useCallback(
+    (property: string, value: string) => {
+      objectConfiguration.updateProperty(property, value);
+      forceUpdate();
+    },
+    [objectConfiguration, forceUpdate]
+  );
+
+  const property = properties.get(propertyName);
+  const extraInfos = property.getExtraInfo();
+  return (
+    <ResourceSelector
+      project={project}
+      // $FlowExpectedError
+      resourceKind={extraInfos.size() > 0 ? extraInfos.at(0) : ''}
+      floatingLabelText={property.getLabel()}
+      resourceManagementProps={resourceManagementProps}
+      initialResourceName={property.getValue()}
+      onChange={value => onChangeProperty(propertyName, value)}
+      resourcesLoader={resourcesLoader}
+      fullWidth
+    />
+  );
+};
+
+const Model3DEditor = ({
+  objectConfiguration,
+  project,
+  layout,
+  resourceManagementProps,
+}: EditorProps) => {
+  const forceUpdate = useForceUpdate();
+  const properties = objectConfiguration.getProperties();
+
+  const onChangeProperty = React.useCallback(
+    (property: string, value: string) => {
+      objectConfiguration.updateProperty(property, value);
+      forceUpdate();
+    },
+    [objectConfiguration, forceUpdate]
+  );
+
+  return (
+    <ColumnStackLayout noMargin>
+      <PropertyResourceSelector
+        objectConfiguration={objectConfiguration}
+        propertyName="modelResourceName"
+        project={project}
+        resourceManagementProps={resourceManagementProps}
+      />
+      <Text size="block-title" noMargin>
+        <Trans>Default orientation</Trans>
+      </Text>
+      <ResponsiveLineStackLayout expand noColumnMargin>
+        {['rotationX', 'rotationY', 'rotationZ'].map(propertyName => (
+          <PropertyField
+            objectConfiguration={objectConfiguration}
+            propertyName={propertyName}
+          />
+        ))}
+      </ResponsiveLineStackLayout>
+      <Text size="block-title" noMargin>
+        <Trans>Default size</Trans>
+      </Text>
+      <ResponsiveLineStackLayout expand noColumnMargin>
+        {['width', 'height', 'depth'].map(propertyName => (
+          <PropertyField
+            objectConfiguration={objectConfiguration}
+            propertyName={propertyName}
+          />
+        ))}
+      </ResponsiveLineStackLayout>
+      <ColumnStackLayout noMargin expand>
+        <PropertyCheckbox
+          objectConfiguration={objectConfiguration}
+          propertyName="keepAspectRatio"
+        />
+        <SelectField
+          value={properties.get('materialType').getValue()}
+          floatingLabelText={properties.get('materialType').getLabel()}
+          helperMarkdownText={properties.get('materialType').getDescription()}
+          onChange={(event, index, newValue) => {
+            onChangeProperty('materialType', newValue);
+          }}
+        >
+          <SelectOption
```

Comment:
What about?

- Basic material (no light effect, no lighting effects)
- Realistic material (lighting effects)
- No change (keep realistic or basic materials defined in the model)

?

Reply:
Why not, but it implies that "no change" is the best choice when it may be almost black (at least with the model I used it was).
For models, the second choice has actually no effect on basic light as it doesn't change the attribute if it doesn't exist.
The 2 objects probably need a different description of the choices.

Reply:
> For models, the second choice has actually no effect on basic light as it doesn't change the attribute if it doesn't exist.

Ah right I missed this. So yes, it's a bit different from models. Maybe:

- Apply basic material (do not react to lights, fastest)
- Keep realistic materials, but remove metalness (react to lights, good for most rendering)
- No change (keep realistic or basic materials defined in the model)